### ISSUE TITLE
Race condition for temp dir creation

### ISSUE THREAD

**Issue by ginkel** [adobe/S3Mock#837]
Hi there,

we are running S3Mock embedded using the `S3MockExtension`. I know that this scenario is deprecated. ;-)

After adding a second unit test class using `S3MockExtension` we are regularly seeing root folder creation failures in CI where two threads generate the same tmp file name and only one of them succeeds with `mkdir` in `StoreConfiguration#rootFolder`.

I have prepared a PR that hopefully fixes this issue by using `Files#createTempDirectory`, which is thread-safe.

Regards,
Thilo

```
01:44:23.689 [ForkJoinPool-1-worker-1] INFO  com.adobe.testing.s3mock.store.StoreConfiguration - Successfully created "/tmp/s3mockFileStore1664243063689" as root folder. Will retain files on exit: false
01:44:23.692 [ForkJoinPool-1-worker-8] WARN  org.springframework.boot.web.servlet.context.AnnotationConfigServletWebServerApplicationContext - Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'bucketStore' defined in class path resource [com/adobe/testing/s3mock/store/StoreConfiguration.class]: Unsatisfied dependency expressed through method 'bucketStore' parameter 1; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'bucketRootFolder' defined in class path resource [com/adobe/testing/s3mock/store/StoreConfiguration.class]: Unsatisfied dependency expressed through method 'bucketRootFolder' parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'rootFolder' defined in class path resource [com/adobe/testing/s3mock/store/StoreConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [java.io.File]: Factory method 'rootFolder' threw exception; nested exception is java.lang.IllegalStateException: Root folder could not be created. Path: /tmp/s3mockFileStore1664243063689
```

**Comment by afranken** [comment]
@ginkel thanks again for the PR, I just released [2.7.1](https://github.com/adobe/S3Mock/releases/tag/2.7.1) containing your fix. :)

**Comment by ginkel** [comment]
@afranken Thanks for the speedy release! :-)
